app/src/main/java/com/babylon/wallet/android/presentation/settings/appsettings/AppSettingsViewModel.kt

```diff
@@ -43,6 +48,24 @@ class AppSettingsViewModel @Inject constructor(
                     }
                 }
         }
+        if (BuildConfig.CRASH_REPORTING_ENABLED) {
+            _state.update { settingsUiState ->
+                settingsUiState.copy(
+                    settings = (settingsUiState.settings + SettingsItem.AppSettingsItem.CrashReporting(false)).toPersistentSet()
+                )
+            }
+            viewModelScope.launch {
+                preferencesManager.isCrashReportingEnabled.collect { enabled ->
+                    if (enabled) {
+                        Firebase.crashlytics.deleteUnsentReports()
+                    }
```

Comment:
Can you explain this a bit? Should we also delete unsent reports when disabling crash reporting or not?

Reply:
Even when you have disabled sending reports to crashlytics, they are stored in local database until sharing crashes is enabled. And we don't want to send anything from the period when user had crash reporting turned off. So we delete all the reports just before enabling sending them, so that only crashes that happen when you have reports on will be sent